File: src/main/cpp/com/deepapp/vn/io/AA/A0/AAA0_0100/worker/AAA0_0100_W.cpp
#include "com/deepapp/infrastructure/BaseWorker.h"
#include "com/deepapp/infrastructure/WorkerRegistry.h"
#include <iostream>
#include <sstream>
#include <chrono>
#include <iomanip>

namespace deepapp {
namespace workers {

/**
 * Example Worker: AAA0_0100_W
 * Demonstrates how to create a worker that auto-registers
 */
class AAA0_0100_Worker : public infrastructure::BaseWorker {
public:
    AAA0_0100_Worker() : BaseWorker("AAA0_0100_Worker") {
        std::cout << "[AAA0_0100_Worker] Initialized" << std::endl;
    }

    std::string processTask(const std::string& event_type, 
                           const std::string& payload) override {
        std::cout << "[AAA0_0100_Worker] Processing task:" << std::endl;
        std::cout << "  Event Type: " << event_type << std::endl;
        std::cout << "  Payload: " << payload << std::endl;

        // Example processing logic
        if (event_type == "echo") {
            return createResponse("success", payload);
        } 
        else if (event_type == "process") {
            return processData(payload);
        }
        else if (event_type == "transform") {
            return transformData(payload);
        }
        else {
            return createResponse("unknown_event", 
                "Event type '" + event_type + "' not supported");
        }
    }

    bool canHandle(const std::string& event_type) const override {
        return event_type == "echo" || 
               event_type == "process" || 
               event_type == "transform";
    }

private:
    std::string processData(const std::string& payload) {
        // Example: Convert to uppercase
        std::string result = payload;
        for (char& c : result) {
            c = std::toupper(c);
        }
        return createResponse("processed", result);
    }

    std::string transformData(const std::string& payload) {
        // Example: Add timestamp
        auto now = std::chrono::system_clock::now();
        auto time_t = std::chrono::system_clock::to_time_t(now);
        
        std::ostringstream oss;
        oss << "{\"timestamp\":\"" 
            << std::put_time(std::localtime(&time_t), "%Y-%m-%d %H:%M:%S")
            << "\",\"data\":\"" << payload << "\"}";
        
        return createResponse("transformed", oss.str());
    }

    std::string createResponse(const std::string& status, 
                              const std::string& data) {
        std::ostringstream oss;
        oss << "{"
            << "\"worker\":\"AAA0_0100_W\","
            << "\"status\":\"" << status << "\","
            << "\"data\":\"" << data << "\","
            << "\"timestamp\":" << std::time(nullptr)
            << "}";
        return oss.str();
    }
};

} // namespace workers
} // namespace deepapp

// Auto-register this worker with task ID "AAA0_0100_W"
REGISTER_WORKER(deepapp::workers::AAA0_0100_Worker, "AAA0_0100_W")

File: src/main/python/com/deepapp/vn/io/AA/A0/AAA0_0100/worker/AAA0_0100_W.py
"""
Example Python Worker: AAA0_0100_W
Demonstrates how to create a worker that auto-registers
"""

import json
from datetime import datetime
from com.deepapp.infrastructure.BaseWorker import BaseWorker
from com.deepapp.infrastructure.WorkerRegistry import register_worker


@register_worker("AAA0_0100_W")
class AAA0_0100_Worker(BaseWorker):
    """
    Example Worker: AAA0_0100_W
    Demonstrates basic worker functionality
    """

    def __init__(self):
        super().__init__("AAA0_0100_Worker")
        self.log("Initialized")

    def process_task(self, event_type: str, payload: str) -> str:
        """Process a task and return result"""
        self.log(f"Processing task: event_type='{event_type}', payload='{payload}'")

        if event_type == "echo":
            return self._echo(payload)
        elif event_type == "process":
            return self._process_data(payload)
        elif event_type == "transform":
            return self._transform_data(payload)
        else:
            return self.create_response("unknown_event",
                f"Event type '{event_type}' not supported")

    def can_handle(self, event_type: str) -> bool:
        """Check if this worker can handle the given event type"""
        return event_type in ["echo", "process", "transform"]

    def _echo(self, payload: str) -> str:
        """Echo the payload back"""
        self.log(f"Echoing payload: {payload}")
        return self.create_response("success", payload)

    def _process_data(self, payload: str) -> str:
        """Process data (example: convert to uppercase)"""
        try:
            result = payload.upper()
            self.log(f"Processed data: '{payload}' -> '{result}'")
            return self.create_response("processed", result)
        except Exception as e:
            self.log(f"Error processing data: {e}", "ERROR")
            return self.create_response("error", str(e))

    def _transform_data(self, payload: str) -> str:
        """Transform data (example: add timestamp)"""
        try:
            now = datetime.now()
            result = {
                "timestamp": now.isoformat(),
                "data": payload,
                "processed_by": self.worker_id
            }
            json_result = json.dumps(result, ensure_ascii=False)
            self.log(f"Transformed data: {json_result}")
            return self.create_response("transformed", result)
        except Exception as e:
            self.log(f"Error transforming data: {e}", "ERROR")
            return self.create_response("error", str(e))